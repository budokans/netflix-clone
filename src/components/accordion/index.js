import React, { useState, useContext, createContext } from "react";
import {
  Container,
  Inner,
  Item,
  Title,
  Header,
  Body,
} from "./styles/accordion";

const ToggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  const [activeIndex, setActiveIndex] = useState();

  return (
    <ToggleContext.Provider value={{ activeIndex, setActiveIndex }}>
      <Container {...restProps}>
        <Inner>{children}</Inner>
      </Container>
    </ToggleContext.Provider>
  );
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Header = function AccordionHeader({ index, children, ...restProps }) {
  const { activeIndex, setActiveIndex } = useContext(ToggleContext);
  const isActive = index === activeIndex;

  return (
    <Header
      onClick={() => {
        !isActive ? setActiveIndex(index) : setActiveIndex(null);
      }}
      {...restProps}
    >
      {children}
      {isActive ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ index, children, ...restProps }) {
  const { activeIndex } = useContext(ToggleContext);
  const isActive = index === activeIndex;

  return isActive ? <Body {...restProps}>{children}</Body> : null;
};
