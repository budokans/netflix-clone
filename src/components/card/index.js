import React, { useState, useContext, createContext } from "react";
import {
  Container,
  Group,
  Title,
  Subtitle,
  Text,
  Entities,
  Meta,
  Item,
  Image,
  Feature,
  Content,
  FeatureTitle,
  FeatureClose,
  Maturity,
  FeatureText,
} from "./styles/card";

const FeatureContext = createContext();

export default function Card({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Card.Group = function CardGroup({ children, ...restProps }) {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState(false);
  const [activeContainerIndex, setActiveContainerIndex] = useState(false);
  return (
    <FeatureContext.Provider
      value={{
        showFeature,
        setShowFeature,
        itemFeature,
        setItemFeature,
        activeContainerIndex,
        setActiveContainerIndex,
      }}
    >
      <Group {...restProps}>{children}</Group>
    </FeatureContext.Provider>
  );
};

Card.Title = function CardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Card.Subtitle = function CardSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Card.Text = function CardText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>;
};

Card.Meta = function CardMeta({ children, ...restProps }) {
  return <Meta {...restProps}>{children}</Meta>;
};

Card.Item = function CardItem({
  item,
  containerIndex,
  children,
  ...restProps
}) {
  const {
    setShowFeature,
    setItemFeature,
    activeContainerIndex,
    setActiveContainerIndex,
  } = useContext(FeatureContext);

  const isActiveContainer = containerIndex === activeContainerIndex;

  return (
    <Item
      onClick={() => {
        if (!isActiveContainer) {
          setActiveContainerIndex(containerIndex);
        }
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...restProps}
    >
      {children}
    </Item>
  );
};

Card.Image = function CardImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Card.Feature = function CardFeature({
  category,
  containerIndex,
  children,
  ...restProps
}) {
  const { itemFeature, showFeature, setShowFeature, activeContainerIndex } =
    useContext(FeatureContext);

  const isActiveContainer = containerIndex === activeContainerIndex;

  return showFeature && isActiveContainer ? (
    <Feature
      {...restProps}
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose
          onClick={() => {
            setShowFeature(false);
          }}
        >
          <img src="/images/icons/close.png" alt="Close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? "PG" : itemFeature.maturity}{" "}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null;
};
