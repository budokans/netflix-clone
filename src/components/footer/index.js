import React from "react";
import {
  Container,
  Row,
  Column,
  Title,
  Link,
  Break,
  Text,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Row = function ({ children, ...restProps }) {
  return <Row {...restProps}>{children}</Row>;
};

Footer.Column = function ({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Title = function ({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Footer.Link = function ({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Break = function ({ ...restProps }) {
  return <Break {...restProps} />;
};

Footer.Text = function ({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};
