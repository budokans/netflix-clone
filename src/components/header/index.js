import React, { useState } from "react";
import { Link as ReachRouterLink } from "react-router-dom";
import {
  Background,
  Container,
  Group,
  Logo,
  Link,
  Text,
  ButtonLink,
  FeatureCTA,
  Feature,
  PlayButton,
  Search,
  SearchIcon,
  SearchInput,
  Profile,
  Picture,
  Dropdown,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? (
    <Background {...restProps} data-testid="header-bg">
      {children}
    </Background>
  ) : (
    children
  );
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReachRouterLink to={to}>
      <Logo {...restProps} />
    </ReachRouterLink>
  );
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.Link = function HeaderLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCTA = function HeaderFeatureCTA({ children, ...restProps }) {
  return <FeatureCTA>{children}</FeatureCTA>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive(!searchActive)}
        data-testid="search-click"
      >
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        onChange={(e) => setSearchTerm(e.target.value)}
        placeholder="Search films and shows"
        value={searchTerm}
        active={searchActive}
        data-testid="search-input"
      />
    </Search>
  );
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`images/users/${src}.png`} {...restProps} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};
