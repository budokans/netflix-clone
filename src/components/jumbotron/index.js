import React from "react";
import { Item, Inner, Container } from "./styles/jumbotron";

export default function Jumbotron({
  children,
  direction = "row",
  ...restProps
}) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};
