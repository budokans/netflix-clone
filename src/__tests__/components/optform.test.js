import React from "react";
import { render } from "@testing-library/react";
import { OptForm } from "../../components";

describe("<Optform />", () => {
  it("renders <Optform /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Text>
          Ready to watch Netflix? Enter your email to create or restart your
          membership.
        </OptForm.Text>
        <OptForm.Input placeholder="Email address" />
        <OptForm.Button>Get started</OptForm.Button>
      </OptForm>
    );

    expect(
      getByText(
        "Ready to watch Netflix? Enter your email to create or restart your membership."
      )
    ).toBeTruthy();
    expect(getByPlaceholderText("Email address")).toBeTruthy();
    expect(getByText("Get started")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
