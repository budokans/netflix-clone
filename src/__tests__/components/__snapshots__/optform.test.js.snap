// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Optform /> renders <Optform /> with populated data 1`] = `
<div
  class="opt-form__Container-sc-1k1ktyx-0 bJgfsX"
>
  <p
    class="opt-form__Text-sc-1k1ktyx-3 eMPjQj"
  >
    Ready to watch Netflix? Enter your email to create or restart your membership.
  </p>
  <input
    class="opt-form__Input-sc-1k1ktyx-1 gCclBh"
    placeholder="Email address"
  />
  <button
    class="opt-form__Button-sc-1k1ktyx-2 djmVxc"
  >
    Get started
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
</div>
`;
